src/Suave/Utils/Collections.fs

```diff
@@ -17,10 +17,15 @@ type IDictionary<'b,'a> with
     | false, _ -> Choice2Of2 "Key was not present"
 
 let getFirst (target : NameValueList) (key : string) =
-  match target |> List.tryPick (fun (a, b) -> if a.Equals key then Some b else None) with
+  match target |> List.tryPick (fun (a, b) -> if a.ToLowerInvariant() = key.ToLowerInvariant() then Some b else None) with
```

Comment:
Ah, that makes sense now. The "req.header origin" expression didn't match b/c origin is "Origin".